File: src/main/java/me/nobaboy/nobaaddons/util/data/DungeonBoss.java
package me.nobaboy.nobaaddons.util.data;

public enum DungeonBoss {
    WATCHER("The Watcher"),
    BONZO("Bonzo"),
    SCARF("Scarf"),
    PROFESSOR("Professor"),
    THORN("Thorn"),
    LIVID("Livid"),
    SADAN("Sadan"),
    MAXOR("Maxor"),
    STORM("Storm"),
    GOLDOR("Goldor"),
    NECRON("Necron"),
    WITHER_KING("Wither King"),
    UNKNOWN("");

    final String text;

    DungeonBoss(String text) {
        this.text = text;
    }

    public static DungeonBoss fromChat(String text) {
        for(DungeonBoss boss : DungeonBoss.values()) {
            if(text.contains(boss.text)) return boss;
        }
        return UNKNOWN;
    }
}

File: src/main/kotlin/me/nobaboy/nobaaddons/util/data/DungeonBoss.kt
package me.nobaboy.nobaaddons.util.data

enum class DungeonBoss(val displayName: String) {
    WATCHER("The Watcher"),
    BONZO("Bonzo"),
    SCARF("Scarf"),
    PROFESSOR("Professor"),
    THORN("Thorn"),
    LIVID("Livid"),
    SADAN("Sadan"),
    MAXOR("Maxor"),
    STORM("Storm"),
    GOLDOR("Goldor"),
    NECRON("Necron"),
    WITHER_KING("Wither King"),
    UNKNOWN("");

    companion object {
        fun fromChat(text: String): DungeonBoss {
            for (boss in entries) {
                if (text.contains(boss.displayName)) return boss
            }
            return UNKNOWN
        }
    }
}